src/os/bluestore/BlueFS.cc

```diff
@@ -1314,29 +1332,154 @@ int BlueFS::_read_random(
     dout(20) << __func__ << " reaching (or past) eof, len clipped to 0x"
 	     << std::hex << len << std::dec << dendl;
   }
+  bool read_done = false;
+  if (h->buf.max_prefetch >= len && h->lock.try_lock()) {
+    if (h->track_read(off, len)) {
+      dout(5) << __func__ << " new sequence: ino "
+	      << h->file->fnode.ino << " "
+	      << h->seq_cnt_last << ","
+	      << h->seq_len_last << ","
+	      << (double)h->seq_len_last / h->seq_cnt_last
+	      << dendl;
+    } /*else {
+      dout(5) << __func__ << " sequence: ino "
+	      << h->file->fnode.ino << " "
+	      << h->seq_cnt << ","
+	      << std::hex << h->prev_offset << std::dec
+	      << (double)h->seq_len_last / h->seq_cnt_last
+	      << dendl;
+    }*/
+
+    if (h->seq_cnt > 2) { //FIXME: make configurable
+      dout(5) << __func__ << " buf read: ino "
+	      << h->file->fnode.ino
+	      << " 0x" << std::hex << off << "~" << len << std::dec
+	      << dendl;
+      ret = _read_random_buffered(h, off, len, out);
```

Comment:
i'm having trouble following what the bheavior here is supposed to be here.. there is track_read that remembers the *previous* sequence, and then this here that indicates whether the current recently is more than 2 sequential reads.  Is the idea that if you do _buffered here, then the OS-level prefetching will kick in?